addon/components/file-browser/component.js

```diff
@@ -2,155 +2,344 @@ import Ember from 'ember';
+                name: files[0].name
+            });
+        },
+        selectItem(item) {
+            this.set('renaming', false);
+            if (this.get('selectedItems.length') > 1) {
+                for (var item_ of this.get('selectedItems')) {
+                    item_.set('isSelected', item_ === item);
+                }
+            } else if (this.get('selectedItems.length') ===  1) {
+                if (item.get('isSelected')) {
+                    item.set('isSelected', false);
+                    return;
+                }
+                let otherItem = this.get('selectedItems.firstObject');
+                otherItem.set('isSelected', false);
             }
+            item.set('isSelected', true);
+            this.set('shiftAnchor', item);
         },
-
-        navigateToItem(item) {
-            let breadcrumbs = this.get('breadcrumbs');
-            let index = breadcrumbs.indexOf(item);
-            if (index === -1) {
-                // TODO: Valid to assume item is a child of currentParent?
-                breadcrumbs.pushObject(item);
+        selectMultiple(item, toggle) {
+            this.set('renaming', false);
+            if (toggle) {
+                item.toggleProperty('isSelected');
             } else {
-                let slicedBread = breadcrumbs.slice(0, index + 1);
-                this.set('breadcrumbs', Ember.A(slicedBread));
+                let items = this.get('items');
+                let anchor = this.get('shiftAnchor');
+                if (anchor) {
+                    let max = Math.max(items.indexOf(anchor), items.indexOf(item));
+                    let min = Math.min(items.indexOf(anchor), items.indexOf(item));
+                    for (var item_ of this.get('items')) {
+                        item_.set('isSelected', item_ === item || item_ === anchor || (items.indexOf(item_) > min && items.indexOf(item_) < max));
+                    }
+                }
+                item.set('isSelected', true);
             }
+            Ember.run.next(this, function(){
+                if (this.get('selectedItems.length') === 1) {
+                    this.set('shiftAnchor', item)
+                }
+            });
         },
-
-        navigateUp() {
-            let breadcrumbs = this.get('breadcrumbs');
-            if (breadcrumbs.length === 1) {
-                return;
+        viewItem() {
+            let item = this.get('selectedItems.firstObject');
+            this.sendAction('openFile', item);
+        },
+        openItem(item, qparams) {
+            this.sendAction('openFile', item, qparams);
+        },
+        downloadItem() {
+            let downloadLink = this.get('selectedItems.firstObject.links.download');
+            window.location = downloadLink;
+        },
+        _deleteItem(item, url) {
+            authenticatedAJAX({
+                url: url,
+                type: 'DELETE',
+                xhrFields: {withCredentials: true}
+            })
```

Comment:
I'd like to see this pushed into the model layer. I think we could pretty easily modify the files adapter to handle deletes.